67420: Remove VM card references in Docker sections of virtualmachines.rst (#655)
freenas/freenas-docs: userguide/virtualmachines.rst

   Edit Rancher Password in Raw File Device

> SSH into the Docker Host

Reviewer: s/into/to/

~~~~~~~~~~~~~~~~~~~~~~~~

Find the Docker Host :guilabel:`Com Port` on the

> :menuselection:`Virtual Machines`

Reviewer: Show in entry order: Virtual Machines, pick VM, see Com Port.

> page. It has a value of :samp:`/dev/nmdm{1b}`, where :literal:`{1b}` is

Reviewer: Com port names have the format :samp:...

(I think the variable part is uppercase, please verify.)

> different for each VM.

Reviewer: s/different/unique/

> Use an SSH client to connect to the %brand% server. The %brand%

Reviewer: Simplify:
"Connect to the %brand% server with an SSH client. The"

> :ref:`SSH` service must be running with the

Reviewer: ":ref:`SSH` service must be running with"

> :guilabel:`Login as Root with Password` option set.

Reviewer: ":guilabel:`Login as Root with Password` enabled."

> At the %brand% console prompt, connect to the Rancher Host with

Reviewer: Rancher Host?  Or Docker Host?

> :samp:`cu -l /dev/nmdm{1b}`, replacing :literal:`{1b}` with the

Reviewer: Again, make sure the trailing letters are upper or lower case.

Docker Host :guilabel:`Com Port`.

> If the terminal does not show a :literal:`rancher login:` prompt,

Reviewer: s/not/not immediately/

.. figure:: images/virtual-machines-docker-devices-rawfile.png

   Changing the Docker Host Password

Start the Docker Host
~~~~~~~~~~~~~~~~~~~~~

Go to
:menuselection:`Virtual Machines`
and click
|ui-options| :menuselection:`--> Start`
to boot the Docker Host.

> A Docker Host can take some time to boot. Connect to the Serial Shell

Reviewer: Not clear what a Serial Shell is here.  Is it a guilabel of a button the user should click to connect to the Rancher console?

Author: I tried to use a generic instruction that applies to both master and angulargui branches. I rework it to be angular specific.

during the startup process to view the activity of the Docker Host.

> The :literal:`RancherOS` graphic is shown. Press :kbd:`Enter` to see the

Reviewer: RancherOS should not be a literal here, it is not something the user will type.
"graphic" is probably console graphics, should probably be more clear about that, or just skip mentioning it at all.

:literal:`ClientHost login:` prompt. Log in with the username
:literal:`rancher` and press :kbd:`Enter`. If a custom password was set
in the raw file, enter it now. Otherwise, enter the default password of
:literal:`docker`. The :literal:`[rancher@ClientHost ~]$` prompt is
shown.

> SSH in to the Docker Host

Reviewer: ```suggestion
SSH to the Docker Host
```

~~~~~~~~~~~~~~~~~~~~~~~~~

Go to
:menuselection:`Virtual Machines`,

> find the Docker Host entry, and see :guilabel:`Com Port`. Com port

Reviewer: ```suggestion
find the Docker Host entry, and locate the :guilabel:`Com Port`. Com port
```

again. Installation time varies with processor and network connection
speed. :literal:`[rancher@ClientHost ~]$` is shown when the installation
is finished.

Enter :command:`ifconfig eth0 | grep 'inet addr'` to view the Rancher
IP address. Enter the IP address followed by :literal:`:8080` into a web
browser to connect to the Rancher web interface. For example, if the IP
address is :literal:`10.231.3.208`, enter :literal:`10.231.3.208:8080`
in the browser.

The Rancher web interface takes a few minutes to start. The web browser
might show a connection error while the web interface starts. If a
:literal:`connection has timed out` error is shown, wait one minute and
refresh the page.

When the Rancher web interface loads, click :guilabel:`Add a host` from
the banner across the top of the screen. Verify that
:guilabel:`This site's address` is chosen and click :guilabel:`Save`.

Follow the steps shown in the Rancher web interface and copy the full
:samp:`sudo docker run` command from the text box. Paste it in the
Docker Host shell. The Docker Host will finish configuring Rancher. A
:literal:`[rancher@ClientHost ~]$` prompt is shown when the
configuration is complete.

> Verify that the configuration is complete. Go to the Rancher web

Reviewer: This is redundant.
```suggestion
Go to the Rancher web
```

~~~~~~~~~~~~~~~~~~~~~

Go to
:menuselection:`Virtual Machines`
and click
|ui-options| :menuselection:`--> Start`
to boot the Docker Host.

A Docker Host can take some time to boot. Connect to the Serial Shell
during the startup process to view the activity of the Docker Host.

.. tip:: A :command:`cu -l` command displays as the shell connects to
   the Docker Host Com Port. This command can also be used to connect
   to the Docker Host during an SSH session into the %brand%
   system. The :samp:`nmdm{XY}` value is different for each started
   Docker Host.

> When a message about :literal:`RancherOS` starting appears and the shell

Reviewer: "about RancherOS starting appears" reads weird, "and the shell stops posting new messages" is ambiguous.  Let's break these up and be more clear:
":literal:`RancherOS is starting` is shown." (Be exact on the message shown.)
"Press :kbd:`Enter` to see the :literal:`ClientHost login:` prompt.  Log in with..."

stops posting new messages, press :kbd:`Enter` to see the
:literal:`ClientHost login:` text and continue to log in.

.. _Installing and Configuring the Rancher Server:

Installing and Configuring the Rancher Server
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> In the serial shell, type :literal:`rancher` for the username and press

Reviewer: "serial shell" is ambiguous, and this is a halting sentence.  Rearrange and clarify:
"Enter :literal:`rancher` as the username. Press :kbd:`Enter`.

> :kbd:`Enter`. Enter either the default password :literal:`docker` or the

Reviewer: "either" here makes it sound like either one will work.  This should really be an "if":
"If a custom password was set, enter it now. Otherwise, enter the default password of :literal:`docker`."

custom password created by editing the raw file and press :kbd:`Enter`.

> A :literal:`[rancher@ClientHost ~]$` prompt will appear.

Reviewer: Future tense...
"The :literal:`...` prompt is shown."

Ensure Rancher has functional networking and can :command:`ping` an

> outside website.

Reviewer: We should mention a specific outside site, and it should probably be one we own, like ixsystems.com.  (Check with IT to see if that's okay.)

.. code-block:: none

>    [rancher@ClientHost ~]$ ping -c 5 8.8.8.8

Reviewer: 8.8.8.8 is a magic number here, and is not going to tell the user what is going on.  Are we using a bare IP address because we don't know if DNS is working?

   PING 8.8.8.8 (8.8.8.8): 56 data bytes
   64 bytes from 8.8.8.8: seq=0 ttl=121 time=17.466 ms
   64 bytes from 8.8.8.8: seq=1 ttl=121 time=17.502 ms
   64 bytes from 8.8.8.8: seq=2 ttl=121 time=17.472 ms
   64 bytes from 8.8.8.8: seq=3 ttl=121 time=17.426 ms
   64 bytes from 8.8.8.8: seq=4 ttl=121 time=17.343 ms

   --- 8.8.8.8 ping statistics ---
   5 packets transmitted, 5 packets received, 0% packet loss
   round-trip min/avg/max = 17.343/17.441/17.502 ms

If :command:`ping` returns an error, adjust the VM
:ref:`Network Interface <vms-network-interface>` and reboot the VM.

Download and install the Rancher system with
:command:`sudo docker run -d --restart=unless-stopped -p 8080:8080 rancher/server`.

> If :literal:`Cannot connect to the Docker daemon` shows, run

Reviewer: If/then, although probably okay here.  But it sounds unsure.
"is shown" works better than "shows" here, but probably better to explicitly point out this is an error.
Also, "run" is oddly used.
"If a :literal:`Cannot connect to the Docker daemon` error is shown, enter"

> :command:`sudo dockerd` and try the command again. Installation time

Reviewer: Try *which* command again?

varies with processor and network connection speed.

> When the command prompt is available, enter

Reviewer: "available" is ambiguous.  Shown?  Can be connected to?

:command:`ifconfig eth0 | grep 'inet addr'`. Copy the

> :literal:`inet addr:` IP address. Enter the IP address appended with

Reviewer: Copy the IP address where?
s/appended with/followed by/

:literal:`:8080` into a web browser to connect to the Rancher server

> |web-ui|. For example, if the IP address was :literal:`10.231.3.208`,

Reviewer: |web-ui| is a FreeNAS-specific term.  Don't use that substitution for other things.  Here, just use a plain "web interface".

"was" is past tense, use "is".

> :literal:`10.231.3.208:8080` is entered in a browser.

Reviewer: Be imperative:

```suggestion
enter :literal:`10.231.3.208:8080` in the browser.
```

The Rancher server takes a few minutes to start. The web browser might

> show a connection error while the Rancher |web-ui| starts. If the

Reviewer: Don't use |web-ui| here.

s/If the/If a/

> browser shows a :literal:`connection has timed out` error, wait one

Reviewer: ```suggestion
:literal:`connection has timed out` error is shown, wait one
```

minute and try again.

> When the Rancher |web-ui| loads, click :guilabel:`Add a host`.

Reviewer: Don't use |web-ui|.  It's not a generic term.

> Ensure the radial :guilabel:`This site's address` button is set and

Reviewer: Radial? Set to what?  How about:
"Verify that the :guilabel:`This site's address` value is set to (what?), and click :guilabel:`Save`."

Author: This site's address is a circular fill-in button. Like a checkbox except it isn't a box or shows a check.

click :guilabel:`Save`.

> Follow the instructions that now display and run the

Reviewer: Wordy.
```suggestion
Following the instructions shown, enter
```

> :command:`sudo docker run --rm --privileged -v` command in the Docker

Reviewer: ```suggestion
:command:`sudo docker run --rm --privileged -v` in the Docker
```

> Host Serial shell. :literal:`Launched Rancher Agent:` displays when the

Reviewer: Again, "serial shell" is ambiguous.  
```suggestion
Host Serial shell. :literal:`Launched Rancher Agent:` is shown when the
```

> command is successful.

Reviewer: Be more specific: "when the agent has been started successfully."

> Go to the :guilabel:`Hosts` page of the Rancher |web-ui| to confirm the

Reviewer: Point out that this is a confirmation check. (Short sentences.)
0. Verify that the configuration is complete by:
1. Go to Rancher web interface, click Hosts.
2. Verify presence of (something) (literal? IP address?  hostname?)
3. If present, configuration is complete.

> Docker Host displays in the |web-ui|. Rancher is now configured and

Reviewer: Don't use |web-ui| here.
